### PR TITLE
Fix cleanup pipeline

### PR DESCRIPTION
**Pull request by yosifkit**
`docker ps` has image tags whereas `docker images` does not have tags for images pulled by digest like `tianon/buildkit:0.12@sha256:...`.  This adjusts the "containers" list to also include the image without the tag information so that it can also be ignored when looking for images to delete.

This PR also increases the number of jobs to keep. Since we have 11 arches, one would get discarded if the `all-cleanup` job was triggered.

Log of a failed cleanup:
```console
docker rmi ... tianon/buildkit@sha256:7b3523bb466d58423d2c3e0055b47c3f892d709da91fbd1eecbc0d2521154deb
...
Error response from daemon: conflict: unable to remove repository reference "tianon/buildkit@sha256:7b3523bb466d58423d2c3e0055b47c3f892d709da91fbd1eecbc0d2521154deb" (must force) - container ec9f2062673b is using its referenced image b1f53eedebf1
```

```console
$ docker ps
CONTAINER ID   IMAGE                  COMMAND                  CREATED              STATUS              PORTS           NAMES
9b90cdf42ea7   21ecbf32f51d           "/bin/sh -c 'buildDe…"   About a minute ago   Up About a minute                   nice_goldwasser
ec9f2062673b   tianon/buildkit:0.12   "dockerd-entrypoint.…"   6 weeks ago          Up 6 weeks          2375-2376/tcp   buildx_buildkit_bashbrew-arm64v8

$ # "docker-images" function for just the one image:
$ docker --digests --no-trunc --format '{{- if (eq .Repository "<none>") -}}{{- .ID -}}{{- else -}}{{- .Repository -}}{{- if (eq .Tag "<none>") -}}{{- "@" -}}{{- .Digest -}}{{- else -}}{{- ":" -}}{{- .Tag -}}{{- end -}}{{- end -}}' tianon/buildkit
tianon/buildkit@sha256:7b3523bb466d58423d2c3e0055b47c3f892d709da91fbd1eecbc0d2521154deb

$ # current container list:
$ docker ps --no-trunc --format '{{ .Image }}' | awk '/[:@]/ { print; next } { print $0 ":latest" }'
sha256:21ecbf32f51dbde71e2be2bbf703e31be0b4a581268606d4c595e2ec459d1189
tianon/buildkit:0.12@sha256:7b3523bb466d58423d2c3e0055b47c3f892d709da91fbd1eecbc0d2521154deb
$ # modified container list:
$ docker ps --no-trunc --format '{{ .Image }}' | awk '/:[^@]+@/ {str=$0; sub(":[^@]+@","@", str); print str } /[:@]/ { print; next } { print $0 ":latest" }'
sha256:21ecbf32f51dbde71e2be2bbf703e31be0b4a581268606d4c595e2ec459d1189
tianon/buildkit@sha256:7b3523bb466d58423d2c3e0055b47c3f892d709da91fbd1eecbc0d2521154deb
tianon/buildkit:0.12@sha256:7b3523bb466d58423d2c3e0055b47c3f892d709da91fbd1eecbc0d2521154deb
```

Successfully tested the modified groovy on s390x.

### CODE REVIEW

**Comment by yosifkit** [comment]
I think we may also need to add resolved "external-pins" to `oi-list`, but it doesn't cause a failure right now.

File: infra/cleanup-pipeline.groovy [hunk 1]
```diff
@@ -148,7 +149,9 @@ node(params.TARGET_NODE) {
 					images="$(grep -vE '^(infosiftr/moby|oisupport/[^:@]+)(:|$)|^busybox:.+-builder$' <<<"$images")" || :
 
 					# exclude images of running containers if running from an explicit tag
-					containers="$(docker ps --no-trunc --format '{{ .Image }}' | awk '/[:@]/ { print; next } { print $0 ":latest" }')" || return 1
+					# or if it was pulled/run by tag@digest (docker images does not show the tag but docker ps does)
+					containers="$(docker ps --no-trunc --format '{{ .Image }}' | awk '/:[^@]+@/ {str=$0; sub(":[^@]+@","@", str); print str } /[:@]/ { print; next } { print $0 ":latest" }')" || return 1
+
```
Comment (LaurentGoderre) [inline infra/cleanup-pipeline.groovy:154]: I tried this and it gave me the same result for both. It's probably something I am doing wrong though.

```bash
> docker run --rm -d alpine@sha256:b312e4b0e2c665d634602411fcb7c2699ba748c36f59324457bc17de485f36f6 tail -f > /dev/null  

> docker ps --no-trunc --format '{{ .Image }}' | awk '/[:@]/ { print; next } { print $0 ":latest" }' 

alpine@sha256:b312e4b0e2c665d634602411fcb7c2699ba748c36f59324457bc17de485f36f6
f6648c04cd6c:latest

> docker ps --no-trunc --format '{{ .Image }}' | awk '/:[^@]+@/ {str=$0; sub(":[^@]+@","@", str); print str } /[:@]/ { print; next } { print $0 ":latest" }'

alpine@sha256:b312e4b0e2c665d634602411fcb7c2699ba748c36f59324457bc17de485f36f6
f6648c04cd6c:latest

```

Comment (yosifkit) [reply infra/cleanup-pipeline.groovy:154]: It happens when you pull the image by `tag@digest` (and tag can technically be unrelated to what the digest may be tagged as on the registry).

```console
$ docker run --rm -d --name foo alpine:foo-bar@sha256:b312e4b0e2c665d634602411fcb7c2699ba748c36f59324457bc17de485f36f6 tail -f /dev/null
$ docker ps
CONTAINER ID   IMAGE            COMMAND               CREATED         STATUS         PORTS     NAMES
de2668cda515   alpine:foo-bar   "tail -f /dev/null"   3 seconds ago   Up 2 seconds             foo
$ docker images alpine
REPOSITORY   TAG       IMAGE ID       CREATED       SIZE
alpine       <none>    f6648c04cd6c   6 weeks ago   7.66MB

$ docker ps --no-trunc --format '{{ .Image }}' | awk '/[:@]/ { print; next } { print $0 ":latest" }'
alpine:foo-bar@sha256:b312e4b0e2c665d634602411fcb7c2699ba748c36f59324457bc17de485f36f6
$ docker ps --no-trunc --format '{{ .Image }}' | awk '/:[^@]+@/ {str=$0; sub(":[^@]+@","@", str); print str } /[:@]/ { print; next } { print $0 ":latest" }'
alpine@sha256:b312e4b0e2c665d634602411fcb7c2699ba748c36f59324457bc17de485f36f6
alpine:foo-bar@sha256:b312e4b0e2c665d634602411fcb7c2699ba748c36f59324457bc17de485f36f6
```

Comment (LaurentGoderre) [reply infra/cleanup-pipeline.groovy:154]: Ah gotcha!

**Review by LaurentGoderre — APPROVED** [review]

**Pull request merged**